rfcs/text/0428-type-inheritance-and-forking.md

```diff
@@ -0,0 +1,952 @@
+
+```json
+{
+  // Old, partial updateEntityType message
+  "entityTypeId": "https://blockprotocol.org/@alice/entity-type/person",
+  "schema": {
+    // The properties here are partially applied to the original Entity Type.
+    "properties": {
+      "https://blockprotocol.org/@alice/property-type/birth-date": {
+        "$ref": "https://blockprotocol.org/@alice/property-type/birth-date/v/1"
+      }
+    }
+  }
+}
+```
+
+must instead use a complete schema
+
+```json
+{
+  // New, complete updateEntityType message
+  "entityTypeId": "https://blockprotocol.org/@alice/entity-type/person",
+  "schema": {
+    "$id": "https://blockprotocol.org/@alice/entity-type/person/v/2",
+    "type": "object",
+    "kind": "entityType",
+    "title": "Person",
+    "properties": {
+      "https://blockprotocol.org/@alice/property-type/name/": {
+        "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+      },
+      "https://blockprotocol.org/@alice/property-type/email": {
+        "$ref": "https://blockprotocol.org/@alice/property-type/email/v/1"
+      },
+      "https://blockprotocol.org/@alice/property-type/phone-number": {
+        "$ref": "https://blockprotocol.org/@alice/property-type/phone-number/v/1"
+      },
+      // The update is inlined in the existing schema
+      "https://blockprotocol.org/@alice/property-type/birth-date": {
+        "$ref": "https://blockprotocol.org/@alice/property-type/birth-date/v/1"
+      }
+    }
+  }
+}
+```
+
+given that the original schema was created as follows
+
+```json
+{
+  "$id": "https://blockprotocol.org/@alice/entity-type/person/v/1",
+  "type": "object",
+  "kind": "entityType",
+  "title": "Person",
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/email": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/email/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/phone-number": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/phone-number/v/1"
+    }
+  }
+}
+```
+
+The above change applies to these Block Protocol operations (`updateEntityType` having already been implemented, and the others proposed in accepted RFCs):
+
+- `updateEntityType`
+- `updatePropertyType`
+- `updateLinkType`
+
+which must all make use of complete schemas in place of partial ones. These examples originate from the [Type System RFC](./0352-graph-type-system.md#interfacing-with-types-1).
+
+The `aggregateEntities` operation may need to change its behavior slightly. When calling `aggregateEntities` for a specific entity type, entities that are returned might be of that entity type or subtypes of it when allowing extended types. This is because the entities of the subtype can be coerced to the requested entity type.
```

Comment:
```suggestion
The `queryEntities` operation may need to change its behavior slightly. When calling `queryEntities` for a specific entity type, entities that are returned might be of that entity type or subtypes of it when allowing extended types. This is because the entities of the subtype can be coerced to the requested entity type.
```